src/ui/public/field_editor/field_editor.js

```diff
@@ -57,6 +57,8 @@ uiModules
 
       self.cancel = redirectAway;
       self.save = function () {
+        self.saving = true;
```

Comment:
I'm always wary of flags like this since they can get stuck in an incorrect state. For instance if `indexPattern.save` fails on line 75 `self.saving` will be stuck as `true`.

Instead, what would you think of creating an array of existing field names on controller instantiation (around line 52) and then use that in our template's conditional instead of the live index pattern?

```
      self.existingFieldNames = self.indexPattern.fields.map((field) => {
        return field.name;
      });
```